src/hooks/useMenuItems.jsx

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from "react";
+import getMenuItems from "../api/MenuItemsGetter/getMenuItems";
+
+function useMenuItems(initialCategory = '') {
+    const [category, setCategory] = useState(initialCategory);
+    const [items, setItems] = useState([]);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+         getMenuItems(category)
+            .then(data => {
+                setItems(data);
+            })
+            .catch(err => {
+                setError('Failed to fetch menu items');
+                console.error(err);
+            });
+    }, [category]); 
```

Comment:
You shouldn't fetch product list on every filter change it should be done once on application load